File: src/main/java/iot/tetracubered/configurations/SmartHubConfiguration.java
package iot.tetracubered.configurations;

import io.quarkus.arc.config.ConfigProperties;
import org.eclipse.microprofile.config.inject.ConfigProperty;

@ConfigProperties(prefix = "smart-hub")
public interface SmartHubConfiguration {

    @ConfigProperty(name = "topics")
    TopicsConfiguration topics();
}

File: src/main/kotlin/iot/tetracubered/configurations/SmartHubConfiguration.kt
package iot.tetracubered.configurations

import io.quarkus.arc.config.ConfigProperties
import org.eclipse.microprofile.config.inject.ConfigProperty

@ConfigProperties(prefix = "smart-hub")
interface SmartHubConfiguration {

    @ConfigProperty(name = "messaging-client")
    fun messagingClient(): MessagingClientConfiguration
}